docs: update aot compiler docs

santoshyadavdev opened the pull request:
Fixes #35415

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #35415


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

santoshyadav198613 commented:
Hi @kapunahelewong ,
Can you review please.

kapunahelewong commented on aio/content/guide/aot-compiler.md:
```diff
@@ -43,32 +43,10 @@ Here are some reasons you might want to use AOT.
 
-* **_Just-in-Time_ (JIT)**, which compiles your app in the browser at runtime.
-* **_Ahead-of-Time_ (AOT)**, which compiles your app at build time.
+* **_Just-in-Time_ (JIT)**, was used as default till Angular 8.
```
```suggestion
* **_Just-in-Time_ (JIT)**, was used as default until Angular 8.
```

santoshyadav198613 replied:
Done.

kapunahelewong reviewed (commented):
One small suggestion. We'll still need engineering to review.

AndrewKushnir reviewed (approved):
Thanks for the PR @santoshyadav198613, the change LGTM 👍

santoshyadav198613 commented:
Hi @kapunahelewong ,
Can we get it reviewed, let me know if any changes are required.

gkalpak commented on aio/content/guide/aot-compiler.md:
```diff
@@ -43,32 +43,10 @@ Here are some reasons you might want to use AOT.
-See the [CLI command reference](cli) and [Building and serving Angular apps](guide/build) for more information.
-
-</div>
+AOT compilation is the default when you run the [`ng build`](cli/build) (build only) or [`ng serve`](cli/serve)  (build and serve locally) CLI commands.
```
Doesn't the default depend on your `angular.json` config?

santoshyadav198613 replied:
With new Angular 9 project, AOT is default. Also ng update also makes AOT true while updating to V9.

gkalpak replied:
AOT is by default set to true in the `angular.json` config. This is different than saying that `ng build`/`ng serve` default to using AOT. I would make this clear in the guide (because there are people that will have their configs set to `aot: false` (for whatever reason).

Given that this guide is about AOT compilation (i.e. it is not an introductory/get started quickly guide), I think we should aim for correctness (vs glossing over under-the-hood details).

WDYT?

santoshyadav198613 replied:
Makes sense George, will try to add more info, like we can change this setting from angular.json.

gkalpak replied:
Sorry, I still find this sentence inaccurate. `ng build/serve` do not default to AOT. They use whatever is in `angular.json` (and if nothing is specified, then JIT is used afaict). It is just that the CLI generates configs with `aot: true` by default.

How about something like this (and remove the alert below):

```suggestion
When you run the [`ng build`](cli/build) (build only) or [`ng serve`](cli/serve) (build and serve locally) CLI commands, the type of compilation (JIT or AOT) depends on the value of the `aot` property in your build configuration specified in `angular.json`. By default, `aot` is set to `true` for new CLI apps.

See the [CLI command reference](cli) and [Building and serving Angular apps](guide/build) for more information.
```

santoshyadav198613 replied:
Thanks, will make the changes :)

santoshyadav198613 commented:
Hi @gkalpak ,
I have added as a note that aot can be changed, though I don't know the reason why someone would like to do that. Let me know if this looks ok.

gkalpak commented on aio/content/guide/aot-compiler.md:
```diff
@@ -43,30 +43,14 @@ Here are some reasons you might want to use AOT.
 
-* **_Just-in-Time_ (JIT)**, which compiles your app in the browser at runtime.
-* **_Ahead-of-Time_ (AOT)**, which compiles your app at build time.
+* **_Just-in-Time_ (JIT)**, was used as default until Angular 8.
+* **_Ahead-of-Time_ (AOT)**, which compiles your app and libraries at build time.
```
This change looks strange. For JIT we mention when it was the default and for AOT we mention what it does :smiley:
How about:

```suggestion
* **_Just-in-Time_ (JIT)**, which compiles your app in the browser at runtime. This was the default until Angular 8.
* **_Ahead-of-Time_ (AOT)**, which compiles your app and libraries at build time. This is the default since Angular 9.
```

santoshyadav198613 replied:
sure looks good to me as well

santoshyadav198613 commented:
Done with the changes @gkalpak

santoshyadav198613 commented:
I can see lint error 
The commit message body does not meet the minimum length of 100 characters
is it mandatory now @gkalpak ?

santoshyadavdev committed:
docs: update aot compiler docs

change to aot docs to include changes after angular 9
aot is true for new applications created, also it will
be updated to aot true once app is updated using ng update

santoshyadav198613 commented:
Hi @gkalpak ,
I have made all the changes, can you please take a look now CI is also green now.

gkalpak commented:
Thx, @santoshyadav198613!
(Yes, the 100 chars commit message body requirement is a new one (and it can be annoying for certain types of commits 🤷‍♂️).)
